Author: Daniel Baumann <mail@daniel-baumann.ch>
Description: Correcting wrong LSB functions in sysvinit script (Closes: #740066),
 see https://github.com/lxc/lxc/issues/309,
 https://github.com/lxc/lxc/issues/310,
 and https://github.com/lxc/lxc/issues/311.

--- a/config/init/sysvinit/lxc-containers.in
+++ b/config/init/sysvinit/lxc-containers.in
@@ -23,25 +23,17 @@
 sysconfdir="@SYSCONFDIR@"
 
 # Source function library.
-test ! -r "$sysconfdir"/rc.d/init.d/functions ||
-        . "$sysconfdir"/rc.d/init.d/functions
-
-# provide action() fallback
-if ! type action >/dev/null 2>&1; then
-    # Real basic fallback for sysvinit "action" verbage.
-    action() {
-        echo -n "$1	"
-        shift
-        "$@" && echo "OK" || echo "Failed"
-    }
-fi
+test ! -r /lib/lsb/init-functions ||
+        . /lib/lsb/init-functions
 
 start() {
-    action $"Starting LXC autoboot containers: " @LIBEXECDIR@/lxc/lxc-containers start
+    log_daemon_msg "Starting LXC autoboot containers: "
+    @LIBEXECDIR@/lxc/lxc-containers start
 }
 
 stop() {
-    action $"Stopping LXC containers: " @LIBEXECDIR@/lxc/lxc-containers stop
+    log_daemon_msg "Stopping LXC containers: "
+    @LIBEXECDIR@/lxc/lxc-containers stop
 }
 
 # See how we were called.
--- a/config/init/sysvinit/lxc-net.in
+++ b/config/init/sysvinit/lxc-net.in
@@ -20,25 +20,17 @@
 sysconfdir="@SYSCONFDIR@"
 
 # Source function library.
-test ! -r "$sysconfdir"/rc.d/init.d/functions ||
-        . "$sysconfdir"/rc.d/init.d/functions
-
-# provide action() fallback
-if ! type action >/dev/null 2>&1; then
-    # Real basic fallback for sysvinit "action" verbage.
-    action() {
-        echo -n "$1	"
-        shift
-        "$@" && echo "OK" || echo "Failed"
-    }
-fi
+test ! -r /lib/lsb/init-functions ||
+        . /lib/lsb/init-functions
 
 start() {
-    action $"Starting LXC network bridge: " @LIBEXECDIR@/lxc/lxc-net start
+    log_daemon_msg "Starting LXC network bridge: "
+    @LIBEXECDIR@/lxc/lxc-net start
 }
 
 stop() {
-    action $"Stopping LXC network bridge: " @LIBEXECDIR@/lxc/lxc-net stop
+    log_daemon_msg "Stopping LXC network bridge: "
+    @LIBEXECDIR@/lxc/lxc-net stop
 }
 
 # See how we were called.
